clouddriver-openstack/src/test/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/validators/servergroup/ResizeOpenstackAtomicOperationValidatorSpec.groovy

```diff
@@ -0,0 +1,135 @@
+    credz = new OpenstackCredentials(credentials)
+    errors = Mock(Errors)
+    credentials = Mock(OpenstackNamedAccountCredentials) {
+      _ * getCredentials() >> credz
+    }
+    provider = Mock(AccountCredentialsProvider) {
+      _ * getCredentials(_) >> credentials
+    }
+    validator = new DeployOpenstackAtomicOperationValidator(accountCredentialsProvider: provider)
+  }
+
+  def "Validate no error"() {
+    given:
+    ServerGroupParameters params = new ServerGroupParameters(instanceType: instanceType, image:image, maxSize: maxSize, minSize: minSize, networkId: networkId, poolId: poolId, securityGroups: securityGroups)
+    DeployOpenstackAtomicOperationDescription description = new DeployOpenstackAtomicOperationDescription(account: account, application: application, region: region, stack: stack, freeFormDetails: freeFormDetails, disableRollback: disableRollback, timeoutMins: timeoutMins, serverGroupParameters: params, credentials: credz)
+
+    when:
+    validator.validate([], description, errors)
+
+    then:
+    0 * errors.rejectValue(_,_)
+  }
+
+
+  @Unroll
+  def "Validate create missing required core field - #attribute"() {
+    given:
+    ServerGroupParameters params = new ServerGroupParameters(instanceType: instanceType, image:image, maxSize: maxSize, minSize: minSize, networkId: networkId, poolId: poolId, securityGroups: securityGroups)
+    DeployOpenstackAtomicOperationDescription description = new DeployOpenstackAtomicOperationDescription(account: account, application: application, region: region, stack: stack, freeFormDetails: freeFormDetails, disableRollback: disableRollback, timeoutMins: timeoutMins, serverGroupParameters: params, credentials: credz)
+    if (attribute != 'stack') {
+      description."$attribute" = ''
+    } else {
+      description."$attribute" = '1-2-3'
+    }
+
+    when:
+    validator.validate([], description, errors)
+
+    then:
+    times * errors.rejectValue(_,_)
+
+    where:
+    attribute << ['application', 'stack']
+    times << [2,1]
+  }
+
+  @Unroll
+  def "Validate create missing required template field - #attribute"() {
+    given:
+    ServerGroupParameters params = new ServerGroupParameters(instanceType: instanceType, image:image, maxSize: maxSize, minSize: minSize, networkId: networkId, poolId: poolId, securityGroups: securityGroups)
+    DeployOpenstackAtomicOperationDescription description = new DeployOpenstackAtomicOperationDescription(account: account, application: application, region: region, stack: stack, freeFormDetails: freeFormDetails, disableRollback: disableRollback, timeoutMins: timeoutMins, serverGroupParameters: params, credentials: credz)
+    description.serverGroupParameters."$attribute" = null
+
+    when:
+    validator.validate([], description, errors)
+
+    then:
+    times * errors.rejectValue(_,_)
+
+    where:
+    attribute << ['instanceType', 'image', 'maxSize', 'minSize', 'networkId', 'poolId', 'securityGroups']
+    times << [1,1,3,3,1,1,1]
```

Comment:
Asking for for you to use a more readable syntax for the given library does not make me a fanboy. I spent the afternoon trying to get Spock to correctly mock an interface defined method that was implemented by a synthetic method (getter on property). Not happy about that failing.

P.S. I am a convert